mediator/src/main/java/cas/ypsiliform/mediator/negotiation/AgentProxy.java

```diff
@@ -63,8 +82,9 @@ public void onAgentRemoved()
 
     public Thenable<AgentResponse> sendSolutionProposals(MediatorRequest solution)
     {
-        // TODO implementation missing
-        throw new UnsupportedOperationException();
+        responseAble = new Thenable<>();
```

Comment:
can you add an assert that there is no message currently running?
e.g. assert responseAble == null || responseAble.isResolved() == true : "don't support parallel messages"